pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -45,26 +45,36 @@ func NewMetadata(config *types.InstallConfig) *Metadata {
 	return &Metadata{BaseDomain: config.BaseDomain, PublishStrategy: config.Publish}
 }
 
-// AccountID returns the IBM Cloud account ID associated with the authentication
-// credentials.
-func (m *Metadata) AccountID(ctx context.Context) (string, error) {
+func (m *Metadata) client() (*Client, error) {
+	if m.sessionClient != nil {
+		return m.sessionClient, nil
+	}
+
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
```

Comment:
Since this is an "internal" function, it shouldn't acquire the lock. That should happen via whoever calls `client()`. The reason for that is that the mutex is also protecting other fields from metadata, like `accountID` and `apiKey`.
We should probably go back to what it was before where the exported methods acquire the lock.